contracts/VetoGuard.sol

```diff
@@ -44,20 +29,21 @@ contract VetoGuard is
             uint256 _executionDelayBlocks,
             address _owner,
             address _vetoERC20Voting,
-            address _gnosisSafe
+            address _gnosisSafe // Address(0) == msg.sender
         ) = abi.decode(initializeParams, (uint256, address, address, address));
 
         executionDelayBlocks = _executionDelayBlocks;
         transferOwnership(_owner);
         vetoERC20Voting = IVetoERC20Voting(_vetoERC20Voting);
-        gnosisSafe = IGnosisSafe(_gnosisSafe);
+        gnosisSafe = IGnosisSafe(
+            _gnosisSafe == address(0) ? msg.sender : _gnosisSafe
```

Comment:
what do you mean by Gnosis proxy address? Is that indeterminate while the safe is being created? Just trying to understand what's happening here.

Reply:
Yes - the Gnosis safe address is determined via the bytes data which is used to setup the safe properly. Therefore, we do not know what the safe address is until the deployment is completed - therefore we must work around this by dynamically passing in the gnosis safe.